File: 136.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int, int> c;
        
        for(int i : nums){
            c[i] += 1;
        }
        
        for (auto &it : c){
            if(it.second == 1)
                return it.first;
        }
        return -1;
    }
};

File: 136.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # With extra memory:
        
        intkey = {}
        for i in range(max(nums)+1):
            intkey[i] = 0

        for i in nums:
            intkey[i] +=1
            
        for k, v in intkey.items():
            if v == 1:
                return k